fili-core/src/main/java/com/yahoo/bard/webservice/metadata/DataSourceMetadataService.java

```diff
@@ -62,7 +76,30 @@ public void update(PhysicalTable table, DataSourceMetadata metadata) {
                         )
                 );
 
-        allSegments.computeIfAbsent(table, ignored -> new AtomicReference<>()).set(current);
+        // Group segment interval by every column present in the segment
+        Map<String, Set<Interval>> currentByColumn = new LinkedHashMap<>();
+
+        // Accumulate all intervals by column name
```

Comment:
This is not a blocker of this PR, but it avoids potential bugs if `update` method is broken down into few helper method

1. 
```
        // Group all the segments by the starting date of their interval.
        // Accumulate all the partitions of a segment in a map indexed by their identifier.
        ConcurrentSkipListMap<DateTime, Map<String, SegmentInfo>> currentByTime = metadata.getSegments().stream()
                .collect(
                        Collectors.groupingBy(
                                seg -> seg.getInterval().getStart(),
                                ConcurrentSkipListMap::new,
                                Collectors.mapping(SegmentInfo::new, COLLECTOR)
                        )
                );
```

2. 
```
        // Group segment interval by every column present in the segment
        Map<String, Set<Interval>> currentByColumn = new LinkedHashMap<>();

        // Accumulate all intervals by column name
        for (DataSegment segment : metadata.getSegments()) {
            SegmentInfo segmentInfo = new SegmentInfo(segment);
            for (String column : segmentInfo.getColumnNames()) {
                currentByColumn.computeIfAbsent(column, ignored -> new HashSet<>()).add(segmentInfo.getInterval());
            }
```

3. 
```
        // Simplify interval sets using SimplifiedIntervalList
        Map<String, Set<Interval>> simplifiedByColumn = currentByColumn.entrySet().stream()
                .collect(
                        Collectors.toMap(
                                Map.Entry::getKey,
                                entry -> new HashSet<>(new SimplifiedIntervalList(entry.getValue()))
                        )
                );
```